### ISSUE TITLE
Is it possible to set a global header like "api-token"?

### ISSUE THREAD

**Issue by aufdenpunkt** [neroniaky/angular2-token#21]
Thanks and best,
Chris

**Comment by neroniaky** [comment]
@aufdenpunkt there is no method to set the headers/header names directly. If you would tell me why this needed we could implement it.

**Comment by aufdenpunkt** [comment]
@neroniaky Thanks for your quick response! So the case would be, that only authorized frontends are allowed to make requests to the API in general.

**Comment by ezilocchi** [comment]
I have gone through the same problem. I create this https://github.com/neroniaky/angular2-token/issues/36

**Comment by aufdenpunkt** [comment]
Hi @neroniaky,

are there any news about this topic?

Thanks and best,
Chris

**Comment by neroniaky** [comment]
Hey @aufdenpunkt, I recently implement the` sendHttpRequest()`-method, which accepts Angular2 `RequestOptions`. It takes the supplied `RequestOptions` and merges them with the authentication headers ('access-token' etc.). This should give you the option to specify additional headers. 
Please let me know if this works for you, or if I got your use case wrong 😃 . Thanks!

**Comment by aufdenpunkt** [comment]
Thanks for your quick reply. So this solution is confusing me a little bit. If I'm getting you right, then I have to extend the BaseRequestOptions, right?

Something like this:

`import { Injectable } from '@angular/core'
import { Headers, BaseRequestOptions } from '@angular/http'

@Injectable()
export class ExtendedRequestOptions extends BaseRequestOptions {
    headers: Headers = new Headers({
      'Test-Header': 'abcd'
    })
}`

**Comment by neroniaky** [comment]
I'm answering this from my phone (beware of typos :smile: ) so I can't try it myself right now, but what you should be able to do:

Define your new Header:
```javascript
let customHeaders = new Headers({
     'Test-Header': 'abcd'
});
```

Use this Header in your RequestOptions:
```javascript
let customRequestOptions = new RequestOptions({
     method: RequestMethod.Get,
     url: 'myurl/123',
     headers: customHeaders
});
```

Then use the RequestOptions with sendHttpRequest:
```javascript
this.tokenService.sendHttpRequest(customRequestOptions);
```

This is exactly what the Angular2-Token does too, when you use `.get()` etc.

Hope this helps! :smile:

**Comment by aufdenpunkt** [comment]
Ah okay, thanks. I got it … This would mean, that I have to write my own GET, POST, PUT, etc. methods right? Sounds a little bit uncomfortable to me.

Why not something like this? 😄 
```
constructor(private _tokenService: Angular2TokenService) {
    this._tokenService.init({
        apiPath:                    null,
        …
        requestOptions: {
            headers: {
                'Test-Header': 'abcd'
            }
        },
        …
   });
}
```

**Comment by neroniaky** [comment]
I have not seen the need for this yet, but makes sense. Will try to implement something similar today or tomorrow.

**Comment by neroniaky** [comment]
Hey just published version 0.1.15. This gives you the option to define custom global headers like you suggested. Be carful when overwriting the default options, because devise token auth will refuse requests without 'Content-Type' set.

**Comment by aufdenpunkt** [comment]
Hey, awesome! Many thanks for this! 👍
